src/map/lua/luautils.h

```diff
@@ -137,7 +137,7 @@ namespace luautils
     auto GetZone(uint16 zoneId) -> std::optional<CLuaZone>;
     auto GetNPCByID(uint32 npcid, sol::object const& instanceObj) -> std::optional<CLuaBaseEntity>;
     auto GetMobByID(uint32 mobid, sol::object const& instanceObj) -> std::optional<CLuaBaseEntity>;
-    auto GetEntityByID(uint32 mobid, sol::object const& instanceObj) -> std::optional<CLuaBaseEntity>;
+    auto GetEntityByID(uint32 mobid, sol::object const& instanceObj, bool silenceWarning) -> std::optional<CLuaBaseEntity>;
```

Comment:
is this bool optional? I don't know what sol does here if you request a concrete type instead of a sol::object (which may be nil if you don't provide it), does it default to false?

Reply:
Threw this together very fast and I may have omitted some details, but it looks like a crash?
https://godbolt.org/z/s59b1d4YT

Reply:
f it gets defined with a default value instead, it'll go through safely, I think.
`auto GetEntityByID(uint32 mobid, sol::object const& instanceObj, bool silenceWarning = false)`

Reply:
Unfortunately not (in this cut down example):
```cpp

auto GetEntityByID(bool silenceWarning = false)
{
    std::cout << "GetEntityByID" << " " << silenceWarning;
    return true;
}
```

```
[sol2] An error occurred and has been passed to an error handler: sol: runtime error: stack index 1, expected boolean, received no value: 
stack traceback:
	[C]: in function 'base.GetEntityByID'
	[string "GetEntityByID()"]:1: in main chunk
terminate called after throwing an instance of 'sol::error'
  what():  sol: runtime error: stack index 1, expected boolean, received no value: 
stack traceback:
	[C]: in function 'base.GetEntityByID'
	[string "GetEntityByID()"]:1: in main chunk
```

Reply:
It'll need to be a sol::object which is checked for being non null, and then converted into a bool

Reply:
> 
```cpp
auto GetEntityByID(bool silenceWarning = false)
{
    std::cout << "GetEntityByID" << " " << silenceWarning;
    return true;
}
```

The = part needs to be at the header side, not the cpp side. based on existing examples.

Reply:
Weird that VS will not even compile it for me on the cpp side then, but happily builds when done in the header only.

Reply:
Updated, optional arg seems to function as wanted now.